node/bft/src/helpers/storage.rs

```diff
@@ -494,6 +494,49 @@ impl<N: Network> Storage<N> {
         Ok(missing_transmissions)
     }
 
+    /// Check the validity of a certificate coming from another validator.
+    ///
+    /// It suffices to check that the signers (author and endorsers) are members of the applicable committee
+    /// and that they form a quorum in the committee.
+    /// Under the fundamental fault tolerance assumption of at most `f` (stake of) faulty validators,
+    /// the quorum check on signers guarantees that at least one correct validator
+    /// has ensured the validity of the proposal contained in the certificate,
+    /// either by construction (by the author) or by checking (by an endorser):
+    /// given `N > 0` total stake, and `f` the largest integer `< N/3` (where `/` is exact rational division),
+    /// we have `N >= 3f + 1`, which implies `N - f >= 2f + 1`, which is always `> f`;
+    /// `N - f` is the quorum stake.
+    pub fn check_incoming_certificate(&self, certificate: &BatchCertificate<N>) -> Result<()> {
```

Comment:
I think we need to change the name of either this function or `check_certificate` to make it more clear which one is used when.

Reply:
I agree that we should change the names. I was thinking that we could do that as we refactor and simplify the certificate checks in subsequent PRs.